### PR TITLE
Add intepolte_v2 for Paddle2.0

### PR DESCRIPTION
**Pull request by tink2123**
### PR types
Performance optimization

### PR changes
APIs

### Describe

Update interpolate API

* Support list in scale_factor
* Modify parameter default values

新增interpolate_v2 原因：
修改了scale_factor参数的类型，没有办法和之前的OP做到兼容。旧的模型参数无法被新的OP加载，因此增加一个V2 版本。

### CODE REVIEW

File: python/paddle/nn/functional/common.py [hunk 1]
```diff
@@ -125,18 +126,12 @@ def interpolate(input,
...
               H_out = floor (H_{in} * scale_{factor})
               W_out = floor (W_{in} * scale_{factor})
-          else:
-              align_corners = True
-              input : (N,C,H_in,W_in)
-              output: (N,C,H_out,W_out) where:
-              H_out = round(H_{in} * scale_{factor})
-              W_out = round(W_{in} * scale_{factor})
 
         Bilinear interpolation:
```
Comment (willthefrog) [inline python/paddle/nn/functional/common.py:140]: remove `:alias` from docstring

Comment (willthefrog) [reply python/paddle/nn/functional/common.py:140]: `Variable` -> `Tensor`

Comment (willthefrog) [reply python/paddle/nn/functional/common.py:140]: update code samples to 2.0 API

File: python/paddle/nn/functional/common.py [hunk 2]
```diff
@@ -327,7 +322,10 @@ def interpolate(input,
 
     if align_mode != 0 and align_mode != 1:
```
Comment (willthefrog) [inline python/paddle/nn/functional/common.py:301]: add fast path for dygraph mode

File: python/paddle/nn/functional/common.py [hunk 3]
```diff
@@ -327,7 +322,10 @@ def interpolate(input,
...
         raise ValueError("align_mode can only be 0 or 1")
-
+    if align_corners != 0 and resample == 'NEAREST':
+        raise ValueError(
```
Comment (willthefrog) [inline python/paddle/nn/functional/common.py:304]: also enforce in v2 op.

File: python/paddle/nn/functional/common.py [hunk 4]
```diff
@@ -442,20 +443,225 @@ def _is_list_or_turple_(data):
...
+                                     len(input.shape) - 2, len(input.shape)))
+            for value in scale:
+                if value <= 0:
+                    raise ValueError("Attr(scale) should be greater than zero.")
+            attrs['scale'] = list(map(float, scale))
         else:
             raise TypeError(
-                "Attr(scale)'s type should be float, int or Variable.")
-
+                "Attr(scale)'s type should be float, int, list or Variable.")
```
Comment (willthefrog) [inline python/paddle/nn/functional/common.py]: `Variable` -> `Tensor`

File: python/paddle/nn/functional/common.py [hunk 5]
```diff
@@ -442,20 +443,225 @@ def _is_list_or_turple_(data):
...
+    https://en.wikipedia.org/wiki/Bilinear_interpolation.
+    
+    For details of bicubic interpolation, please refer to Wikipedia:
+    https://en.wikipedia.org/wiki/Bicubic_interpolation
+    
+    For details of trilinear interpolation, please refer to Wikipedia:
+    https://en.wikipedia.org/wiki/Trilinear_interpolation.
+    
+    Parameters:
+        input (Variable): 3-D, 4-D or 5-D Tensor, its data type is float32, float64, or uint8,
```
Comment (willthefrog) [inline python/paddle/nn/functional/common.py]: `Variable`

File: python/paddle/nn/functional/common.py [hunk 6]
```diff
@@ -442,20 +443,225 @@ def _is_list_or_turple_(data):
...
+             align_mode=0,
+             data_format='NCHW',
+             name=None):
+    """
+    This op resizes a batch of images.
+    The input must be a 3-D Tensor of the shape (num_batches, channels, in_w)
+    or 4-D (num_batches, channels, in_h, in_w), or a 5-D Tensor of the shape
+    (num_batches, channels, in_d, in_h, in_w) or (num_batches, in_d, in_h, in_w, channels),
+    and the resizing only applies on the three dimensions(depth, height and width).
+    **Warning:** the parameter :attr:`actual_shape` will be deprecated in the
```
Comment (willthefrog) [inline python/paddle/nn/functional/common.py]: remove

File: python/paddle/nn/functional/common.py [hunk 7]
```diff
@@ -442,20 +443,225 @@ def _is_list_or_turple_(data):
...
+    
+    Bicubic interpolation is an extension of cubic interpolation for interpolating
+    data points on a two-dimensional regular grid. The interpolated surface is
+    smoother than corresponding surfaces obtained by bilinear interpolation or
+    nearest-neighbor interpolation.
+    Trilinear interpolation is an extension of linear interpolation for
+    interpolating functions of three variables (e.g. D-direction,
+    H-direction and W-direction in this op) on a rectilinear 3D grid.
+    The linear interpolation is performed on three directions.
+    Align_corners and align_mode are optional parameters,the calculation method
```
Comment (willthefrog) [inline python/paddle/nn/functional/common.py]: `align_corners`

File: python/paddle/nn/functional/common.py [hunk 8]
```diff
@@ -442,20 +443,225 @@ def _is_list_or_turple_(data):
...
+        ValueError: size length should be 3 for input 5-D tensor.
+        ValueError: scale_factor should be greater than zero.
+        TypeError: align_corners should be a bool value
+        ValueError: align_mode can only be '0' or '1'
+        ValueError: data_format can only be 'NCW', 'NWC', 'NCHW', 'NHWC', 'NCDHW' or 'NDHWC'.
+    Examples:
+        .. code-block:: python
+            import paddle
+            import numpy as np
+            import paddle.fluid.dygraph as dg
```
Comment (willthefrog) [inline python/paddle/nn/functional/common.py]: update to 2.0 API

File: python/paddle/nn/layer/common.py [hunk 9]
```diff
@@ -268,6 +270,163 @@ def forward(self, input):
         return out
 
 
+class UpsamplingNearest2d(layers.Layer):
+    """
+    This op upsamples a batch of images, using nearest neighbours' pixel values.
+    The input must be a 4-D Tensor of the shape (num_batches, channels, in_h, in_w), 
+    and the upsampling only applies on the two dimensions(height and width).
+    **Warning:** the parameter :attr:`actual_shape` will be deprecated in the
```
Comment (willthefrog) [inline python/paddle/nn/layer/common.py]: remove

File: python/paddle/nn/layer/common.py [hunk 10]
```diff
@@ -268,6 +270,163 @@ def forward(self, input):
...
+            input_data = np.random.rand(2,3,6,10).astype("float32")
+            place = paddle.fluid.CPUPlace()
+            with dg.guard(place) as g:
+                input = dg.to_variable(input_data)
+                output = upsample_op(input=input)
+                print(output.shape)
+                # [2L, 3L, 12L, 12L]
+    """
+
+    def __init__(self, size=None, scale_factor=None, data_format='NCHW'):
```
Comment (willthefrog) [inline python/paddle/nn/layer/common.py]: `name`

File: python/paddle/nn/layer/common.py [hunk 11]
```diff
@@ -268,6 +270,163 @@ def forward(self, input):
...
+        super(UpSample, self).__init__()
+        self.size = size
+        self.scale_factor = scale_factor
+        self.mode = mode.lower()
+        self.data_format = data_format
+
+    def forward(self, input):
```
Comment (willthefrog) [inline python/paddle/nn/layer/common.py]: `input` -> `x`

File: python/paddle/nn/functional/common.py [hunk 12]
```diff
@@ -442,20 +443,225 @@ def _is_list_or_turple_(data):
...
     helper.append_op(
-        type='{}_interp'.format(resample_type),
+        type='{}_interp_v2'.format(resample_type),
         inputs=inputs,
         outputs={"Out": out},
         attrs=attrs)
     return out
 
 
+def upsample(input,
```
Comment (willthefrog) [inline python/paddle/nn/functional/common.py]: `input` -> `x`

**Review by willthefrog — CHANGES_REQUESTED** [review]

File: python/paddle/nn/functional/common.py [hunk 13]
```diff
@@ -250,53 +242,16 @@ def interpolate(input,
...
-    		input = dg.to_variable(input_data)
-    		output = paddle.nn.functional.interpolate(input=input, size=[12,12])
-    		print(output.shape)
-		# [2L, 3L, 12L, 12L]
+            import paddle.nn.functional as F
+            paddle.disable_static()
+            
+            input_data = np.random.rand(2,3,6,10).astype("float32")
+            x = paddle.to_tensor(input_data)
+            output = F.interpolate(x=x, size=[12,12])
```
Comment (LDOUBLEV) [inline python/paddle/nn/functional/common.py]: 是否需要多举个例子说明不同mode的区别

Comment (tink2123) [reply python/paddle/nn/functional/common.py]: Done~ thx!

File: python/paddle/nn/functional/common.py (hunk 12 above)
Comment (tink2123) [reply python/paddle/nn/functional/common.py]: Above problems has been fixed, please review again

File: python/paddle/nn/functional/common.py [hunk 14]
```diff
@@ -449,20 +419,245 @@ def _is_list_or_turple_(data):
...
+    
+    Bicubic interpolation is an extension of cubic interpolation for interpolating
+    data points on a two-dimensional regular grid. The interpolated surface is
+    smoother than corresponding surfaces obtained by bilinear interpolation or
+    nearest-neighbor interpolation.
+    Trilinear interpolation is an extension of linear interpolation for
+    interpolating functions of three variables (e.g. D-direction,
+    H-direction and W-direction in this op) on a rectilinear 3D grid.
+    The linear interpolation is performed on three directions.
+    Align_corners and align_mode are optional parameters,the calculation method
```
Comment (willthefrog) [inline python/paddle/nn/functional/common.py]: Align -> align

Comment (tink2123) [reply python/paddle/nn/functional/common.py]: Done, thx~

**Review by jzhang533 — DISMISSED** [review]
lgtm

**Review by willthefrog — DISMISSED** [review]

**Review by willthefrog — APPROVED** [review]
LGTM

**Review by jzhang533 — APPROVED** [review]
lgtm

**Review by XiaoguangHu01 — APPROVED** [review]
LGTM

**Comment by luotao1** [comment]
```
2020-08-27 14:24:06 1. It is an Op accuracy problem, please take care of it. You must have one RD (juncaipeng (Recommend), zhangting2020 (Recommend) or luotao1) approval for the python/paddle/fluid/tests/unittests/white_list/op_accuracy_white_list.py, which manages the white list of upgrading the precision of op test to float64. For more information, please refer to: https://github.com/PaddlePaddle/Paddle/wiki/Upgrade-OP-Precision-to-Float64. 
2020-08-27 14:24:06 2. It is an Op accuracy problem, please take care of it. You must have one RD (juncaipeng (Recommend), zhangting2020 or luotao1) approval for the python/paddle/fluid/tests/unittests/white_list/op_threshold_white_list.py, which manages the white list of error threshold for op test with float64 precision. For more information, please refer to: https://github.com/PaddlePaddle/Paddle/wiki/Upgrade-OP-Precision-to-Float64. 
2020-08-27 14:24:06 5. Please use the default precision parameters of 'atol, rtol, eps, max_relative_error'. If you don't use the default value, you must have one RD (Xreki (Recommend), luotao1, lanxianghit or phlrain) approval for the usage of other values. The detailed information is in the link: https://github.cor/PaddlePaddle/Paddle/wiki/OP-test-accuracy-requirements. The error line is 
2020-08-27 14:24:06 +            self.check_output(atol=1e-7)
2020-08-27 14:24:06 +            self.check_output(atol=1e-5)
2020-08-27 14:24:06 +        self.check_output_with_place(place=core.CPUPlace(), atol=1)
2020-08-27 14:24:06 +            self.check_output_with_place(place=core.CPUPlace(), atol=1e-7)
2020-08-27 14:24:06 +            self.check_output_with_place(place=core.CPUPlace(), atol=1e-5)
2020-08-27 14:24:06 +        self.check_output_with_place(place=core.CPUPlace(), atol=1)
2020-08-27 14:24:06 +        self.check_output_with_place(place=core.CPUPlace(), atol=1)
2020-08-27 14:24:06 There are 6 approved errors.
2020-08-27 14:24:06 ****************
```
@XiaoguangHu01 @lanxianghit 因为这个OP不符合精度规范，请问可以豁免么？
@tink2123 可以说下原因

**Comment by tink2123** [comment]
> ```
> 2020-08-27 14:24:06 1. It is an Op accuracy problem, please take care of it. You must have one RD (juncaipeng (Recommend), zhangting2020 (Recommend) or luotao1) approval for the python/paddle/fluid/tests/unittests/white_list/op_accuracy_white_list.py, which manages the white list of upgrading the precision of op test to float64. For more information, please refer to: https://github.com/PaddlePaddle/Paddle/wiki/Upgrade-OP-Precision-to-Float64. 
> 2020-08-27 14:24:06 2. It is an Op accuracy problem, please take care of it. You must have one RD (juncaipeng (Recommend), zhangting2020 or luotao1) approval for the python/paddle/fluid/tests/unittests/white_list/op_threshold_white_list.py, which manages the white list of error threshold for op test with float64 precision. For more information, please refer to: https://github.com/PaddlePaddle/Paddle/wiki/Upgrade-OP-Precision-to-Float64. 
> 2020-08-27 14:24:06 5. Please use the default precision parameters of 'atol, rtol, eps, max_relative_error'. If you don't use the default value, you must have one RD (Xreki (Recommend), luotao1, lanxianghit or phlrain) approval for the usage of other values. The detailed information is in the link: https://github.cor/PaddlePaddle/Paddle/wiki/OP-test-accuracy-requirements. The error line is 
> 2020-08-27 14:24:06 +            self.check_output(atol=1e-7)
> 2020-08-27 14:24:06 +            self.check_output(atol=1e-5)
> 2020-08-27 14:24:06 +        self.check_output_with_place(place=core.CPUPlace(), atol=1)
> 2020-08-27 14:24:06 +            self.check_output_with_place(place=core.CPUPlace(), atol=1e-7)
> 2020-08-27 14:24:06 +            self.check_output_with_place(place=core.CPUPlace(), atol=1e-5)
> 2020-08-27 14:24:06 +        self.check_output_with_place(place=core.CPUPlace(), atol=1)
> 2020-08-27 14:24:06 +        self.check_output_with_place(place=core.CPUPlace(), atol=1)
> 2020-08-27 14:24:06 There are 6 approved errors.
> 2020-08-27 14:24:06 ****************
> ```
> 
> @XiaoguangHu01 @lanxianghit 因为这个OP不符合精度规范，请问可以豁免么？
> @tink2123 可以说下原因

这个问题应该是因为 numpy 和 c++ 在非整数计算插值的时候精度达不到要求。这次没有改旧的OP，之前的版本一直就在白名单里，新版本也需要加上才能过单测。对比过pytorch和paddle的输出结果，精度误差在1e-6可以对齐。

**Comment by XiaoguangHu01** [comment]
> > ```
> > 2020-08-27 14:24:06 1. It is an Op accuracy problem, please take care of it. You must have one RD (juncaipeng (Recommend), zhangting2020 (Recommend) or luotao1) approval for the python/paddle/fluid/tests/unittests/white_list/op_accuracy_white_list.py, which manages the white list of upgrading the precision of op test to float64. For more information, please refer to: https://github.com/PaddlePaddle/Paddle/wiki/Upgrade-OP-Precision-to-Float64. 
> > 2020-08-27 14:24:06 2. It is an Op accuracy problem, please take care of it. You must have one RD (juncaipeng (Recommend), zhangting2020 or luotao1) approval for the python/paddle/fluid/tests/unittests/white_list/op_threshold_white_list.py, which manages the white list of error threshold for op test with float64 precision. For more information, please refer to: https://github.com/PaddlePaddle/Paddle/wiki/Upgrade-OP-Precision-to-Float64. 
> > 2020-08-27 14:24:06 5. Please use the default precision parameters of 'atol, rtol, eps, max_relative_error'. If you don't use the default value, you must have one RD (Xreki (Recommend), luotao1, lanxianghit or phlrain) approval for the usage of other values. The detailed information is in the link: https://github.cor/PaddlePaddle/Paddle/wiki/OP-test-accuracy-requirements. The error line is 
> > 2020-08-27 14:24:06 +            self.check_output(atol=1e-7)
> > 2020-08-27 14:24:06 +            self.check_output(atol=1e-5)
> > 2020-08-27 14:24:06 +        self.check_output_with_place(place=core.CPUPlace(), atol=1)
> > 2020-08-27 14:24:06 +            self.check_output_with_place(place=core.CPUPlace(), atol=1e-7)
> > 2020-08-27 14:24:06 +            self.check_output_with_place(place=core.CPUPlace(), atol=1e-5)
> > 2020-08-27 14:24:06 +        self.check_output_with_place(place=core.CPUPlace(), atol=1)
> > 2020-08-27 14:24:06 +        self.check_output_with_place(place=core.CPUPlace(), atol=1)
> > 2020-08-27 14:24:06 There are 6 approved errors.
> > 2020-08-27 14:24:06 ****************
> > ```
> > 
> > 
> > @XiaoguangHu01 @lanxianghit 因为这个OP不符合精度规范，请问可以豁免么？
> > @tink2123 可以说下原因
> 
> 这个问题应该是因为 numpy 和 c++ 在非整数计算插值的时候精度达不到要求。这次没有改旧的OP，之前的版本一直就在白名单里，新版本也需要加上才能过单测。对比过pytorch和paddle的输出结果，精度误差在1e-6可以对齐。

确认可以先加入白名单。跟之前的op保持一致。@luotao1

**Review by raindrops2sea — APPROVED** [review]
增补一个pr注释说明一下新增v2 op的考量。

**Review by luotao1 — APPROVED** [review]
LGTM

**Pull request merged**